### PR TITLE
Fix vroute.c on macos

### PR DESCRIPTION
**Pull request by tobhe**
- Enable FreeBSD default route workaround for macos
- Fix route cleanup on macos

### CODE REVIEW

File: iked/vroute.c [hunk 1]
```diff
@@ -203,21 +208,34 @@ vroute_insertroute(struct iked *env, int rdomain, struct sockaddr *dest)
...
+vroute_removeroute(struct iked *env, int rdomain, struct sockaddr *dest,
+    struct sockaddr *mask)
 {
 	struct iked_vroute_sc	*ivr = env->sc_vroute;
 	struct vroute_route	*route;
 
 	TAILQ_FOREACH(route, &ivr->ivr_routes, vr_entry) {
 		if (sockaddr_cmp(dest, (struct sockaddr *)&route->vr_dest, -1))
 			continue;
+		if (mask && !(route->vr_flags | RTA_NETMASK))
```
Comment (bluerise) [inline iked/vroute.c:234]: Should that be &?

File: iked/vroute.c [hunk 2]
```diff
@@ -203,21 +208,34 @@ vroute_insertroute(struct iked *env, int rdomain, struct sockaddr *dest)
...
+			continue;
+		if (mask &&
+		    sockaddr_cmp(mask, (struct sockaddr *)&route->vr_mask, -1))
```
Comment (bluerise) [inline iked/vroute.c:237]: Usually I'd put the call in the first line, and one of the arguments in the second, but maybe what you did is better for readability.

File: iked/vroute.c [hunk 3]
```diff
@@ -408,6 +425,10 @@ vroute_getroute(struct iked *env, struct imsg *imsg)
 		return (vroute_dodefaultroute(env, flags, addrs, type, gateway));
 	}
 #endif
+	if (type == RTM_ADD)
```
Comment (bluerise) [inline iked/vroute.c:428]: That's new. Was there no vroute_insertroute before? Was it missing? Or was another call moved?

File: iked/vroute.c [hunk 4]
```diff
@@ -602,6 +624,13 @@ vroute_dodefaultroute(struct iked *env, int flags, int addrs, uint8_t type,
 	default:
 		return (-1);
 	}
+
+	if (type == RTM_ADD)
```
Comment (bluerise) [inline iked/vroute.c:628]: Where does this come from? Was it missing before? Same question for the block below.

**Review by bluerise — CHANGES_REQUESTED** [review]

File: iked/vroute.c (hunk 3 above)
Comment (tobhe) [reply iked/vroute.c:428]: was missing

**Review by bluerise — APPROVED** [review]

**Pull request merged**